src/biogeophys/CanopyFluxesMod.F90

```diff
@@ -634,15 +678,104 @@ subroutine CanopyFluxes(bounds,  num_exposedvegp, filter_exposedvegp,
+            ! otherwise calculate dbh from stem biomass
+            if(use_cn) then
+               if(stem_biomass(p) > 0._r8) then 
+                  dbh(p) = 2._r8 * sqrt(stem_biomass(p) * (1._r8 - fbw(patch%itype(p))) &
+                       / ( shr_const_pi * htop(p) * k_cyl_vol & 
+                       * nstem(patch%itype(p)) *  wood_density(patch%itype(p))))
+               else
+                  dbh(p) = 0._r8
+               endif
+            else
+               dbh(p) = dbh_param(patch%itype(p))
+            endif
+
+            ! leaf and stem surface area
+            sa_leaf(p) = elai(p)
+            ! double in spirit of full surface area for sensible heat
+            sa_leaf(p) = 2.*sa_leaf(p)
+
+            ! Surface area for stem
+            sa_stem(p) = nstem(patch%itype(p))*(htop(p)*shr_const_pi*dbh(p))
+            ! adjust for departure of cylindrical stem model
+            sa_stem(p) = k_cyl_area * sa_stem(p)
+
+            !
+            ! only calculate separate leaf/stem heat capacity for trees
+            ! and shrubs if dbh is greater than some minimum value
+            ! (set surface area for stem, and fraction absorbed by stem to zero)
+            if(.not.(is_tree(patch%itype(p)) .or. is_shrub(patch%itype(p))) &
+                 .or. dbh(p) < min_stem_diameter) then
+               frac_rad_abs_by_stem(p) = 0.0
+               sa_stem(p) = 0.0
+            endif
+
+            ! cross-sectional area of stems
+            carea_stem = shr_const_pi * (dbh(p)*0.5)**2
+
+            ! if using Satellite Phenology mode, calculate leaf and stem biomass
+            if(.not. use_cn) then
+               ! boreal needleleaf lma*c2b ~ 0.25 kg dry mass/m2(leaf)
+               leaf_biomass(p) = 0.25_r8 * max(0.01_r8, sa_leaf(p)) &
+                    / (1.-fbw(patch%itype(p)))
+               stem_biomass(p) = carea_stem * htop(p) * k_cyl_vol &
+                    * nstem(patch%itype(p)) * wood_density(patch%itype(p)) &
+                    /(1.-fbw(patch%itype(p)))
+            endif
+
+            ! internal longwave fluxes between leaf and stem
+            ! (use same area of interaction i.e. ignore leaf <-> leaf)
+            sa_internal(p) = min(sa_leaf(p),sa_stem(p))
+            sa_internal(p) = k_internal * sa_internal(p)
+
+            ! calculate specify heat capacity of vegetation
+            ! as weighted averaged of dry biomass and water
+            ! lma_dry has units of kg dry mass/m2 here
+            ! (Appendix B of Bonan et al., GMD, 2018) 
+
+            cp_leaf(p)  = leaf_biomass(p) * (c_dry_biomass*(1.-fbw(patch%itype(p))) + (fbw(patch%itype(p)))*c_water)
+
+            ! cp-stem will have units J/k/ground_area
+            cp_stem(p) = stem_biomass(p) * (c_dry_biomass*(1.-fbw(patch%itype(p))) + (fbw(patch%itype(p)))*c_water)
+            ! adjust for departure from cylindrical stem model
+            cp_stem(p) = k_cyl_vol * cp_stem(p)
```

Comment:
@swensosc I wasn't expecting this conversion factor for the heat capacity calculation.  It seemed more of a geometry thing (affecting fluxes) and not a mass thing (affecting heat capacity).  Is it a moot point because this term propogates to the desired calculation either way?

Reply:
@rgknox @swensosc responded below. But, I'm putting his answer here so to connect to this conversation.

> are you referring to this:
> 
> cp_stem(p) = k_cyl_vol * cp_stem(p)

Reply:
Note currently k_cyl_vol is 1, so it doesn't change it. The description is that this is departure from having a perfect cylindrical stem.

Reply:
Resolving this conversation so this PR can be resolved.

Reply:
Resolving this so the discussion should go in #1247.